runtime/src/accounts_file.rs

```diff
@@ -0,0 +1,182 @@
+use {
+    crate::{
+        accounts_file::AccountsFileEntry::AppendVecFile, append_vec::*,
+        storable_accounts::StorableAccounts,
+    },
+    solana_sdk::{account::ReadableAccount, clock::Slot, hash::Hash, pubkey::Pubkey},
+    std::{
+        borrow::Borrow,
+        io,
+        path::{Path, PathBuf},
+    },
+};
+
+#[derive(Debug)]
+/// An enum for accessing an accounts file which can be implemented
+/// under different formats.
+pub enum AccountsFileEntry {
```

Comment:
Yep, I don't like the current name as well xD, but the ideal name is already occupied xD.

Essentially it will live under `AccountsStorageEntry`, and each instance handles a file.  Maybe calling it `AccountsFile` or `AccountsFileHandle`, or `AccountsStorageHandle`?

Reply:
Let's just call it AccountsFile.  The shortest!